neuralhydrology/modelzoo/transformer.py

```diff
@@ -0,0 +1,167 @@
+                                              dropout=cfg.transformer_positional_dropout, 
+                                              max_len=cfg.seq_length)
+
+        # positional mask
+        self.mask = None
+
+        # encoder
+        encoder_layers = nn.TransformerEncoderLayer(d_model=self.encoder_dim, 
+                                                    nhead=cfg.transformer_nheads, 
+                                                    dim_feedforward=cfg.transformer_dim_feedforward, 
+                                                    dropout=cfg.transformer_dropout)
+        self.encoder = nn.TransformerEncoder(encoder_layer=encoder_layers, 
+                                             num_layers=cfg.transformer_nlayers,
+                                             norm=None) 
+
+        # head (instead of a decoder)
+        self.dropout = nn.Dropout(p=cfg.output_dropout)
+        self.head = get_head(cfg=cfg, n_in=self.encoder_dim, n_out=self.output_size)
+
+        # init weights and biases 
+        self._reset_parameters()
+
+    def _reset_parameters(self): 
+        initrange = 0.1
+        for layer in self.encoder.layers:
+            layer.linear1.weight.data.uniform_(-initrange, initrange)
+            layer.linear1.bias.data.zero_()
+            layer.linear2.weight.data.uniform_(-initrange, initrange)
+            layer.linear2.bias.data.zero_()
+        self.embedding.weight.data.uniform_(-initrange, initrange)
+        self.embedding.bias.data.zero_()
+
+    def forward(self, data: Dict[str, torch.Tensor], **kwargs) -> Dict[str, torch.Tensor]:
+        """Perform a forward pass on a transformer model without decoder.
+
+        Parameters
+        ----------
+        data : Dict[str, torch.Tensor]
+            Dictionary, containing input features as key-value pairs.
+
+        Returns
+        -------
+        Dict[str, torch.Tensor]
+            Model outputs and intermediate states as a dictionary. 
+                - `y_hat`: model predictions of shape [batch size, sequence length, number of target variables].
+        """
+        # transpose to [seq_length, batch_size, n_features]
+        x_d = data['x_d'].transpose(0, 1)
+
+        # concat all inputs
+        if 'x_s' in data and 'x_one_hot' in data:
+            x_s = data['x_s'].unsqueeze(0).repeat(x_d.shape[0], 1, 1)
+            x_one_hot = data['x_one_hot'].unsqueeze(0).repeat(x_d.shape[0], 1, 1)
+            x_d = torch.cat([x_d, x_s, x_one_hot], dim=-1)
+        elif 'x_s' in data:
+            x_s = data['x_s'].unsqueeze(0).repeat(x_d.shape[0], 1, 1)
+            x_d = torch.cat([x_d, x_s], dim=-1)
+        elif 'x_one_hot' in data:
+            x_one_hot = data['x_one_hot'].unsqueeze(0).repeat(x_d.shape[0], 1, 1)
+            x_d = torch.cat([x_d, x_one_hot], dim=-1)
+        else:
+            pass
+
+        # embedding
+        x_d = self.embedding(x_d) * math.sqrt(self.embedding_dim)        
```

Comment:
A bit nit-picky: This seems to be the only purpose of `self.embedding_dim` outside of the `__init__` (where you could also initialize all layers with directly from the config), so why don't you compute `math.sqrt(embedding_dim)` once in the `__init__` and don't recompute it here all the time. It doesn't change much but is a bit cleaner.